internal/events/aggregator.go

```diff
@@ -638,6 +647,9 @@ func (ag *aggregator) readyForDispatch(ctx context.Context, msg *core.Message, d
 		} else if valid {
 			action = core.ActionConfirm
 		}
+
+	default:
+		action = core.ActionConfirm
```

Comment:
Unrelated find - we were rejecting messages with 0 data elements simply because the default `action` was 0, which means "reject". That didn't seem correct - if there were no special conditions to parse, I think the default should be "confirm" (which includes messages that don't carry data).

If we specifically want to reject messages that contain no data items, it should be a separate branch with an explicit reject reason.

Reply:
I think we might have a bug reference for this - would be good to mark that as resolved... will see if I can find it.

Reply:
Found this instead, which might bear you taking a quick look to see if now is the time to address it (in 1.3, but not this PR): https://github.com/hyperledger/firefly/issues/1270

... but I couldn't find the send-empty-message bug report. So maybe I imagined it.

Reply:
You're likely thinking of https://github.com/hyperledger/firefly/issues/1127, which was resolved - but I think this crept back in as a different flavor of the same issue during one of the refactors in this file.